src/Layer/LabelLayer.js

```diff
@@ -248,19 +249,32 @@ class LabelLayer extends Layer {
 
                 // Necessary event listener, to remove any Label attached to
                 // this tile
-                node.addEventListener('removed', () => {
-                    result.forEach(labels => labels.forEach(label => node.remove(label)));
-                    const child = node.domElements[this.id].dom;
-                    child.parentElement.removeChild(child);
-                });
+                node.addEventListener('removed', () => (this.removeLabelsFromNode(node, result)));
             }
 
             node.layerUpdateState[this.id].noMoreUpdatePossible();
         });
     }
 
+    removeLabelsFromNode(node, labelsArray) {
+        if (node.domElements[this.id]) {
+            labelsArray.forEach(labels => labels.forEach(l => node.remove(l)));
+            const child = node.domElements[this.id].dom;
+            child.parentElement.removeChild(child);
+        }
+    }
+
     delete() {
         this.domElement.parentElement.removeChild(this.domElement);
+
+        for (const root of this.parent.level0Nodes) {
+            root.traverse((obj) => {
+                if (obj.domElements && obj.domElements[this.id]) {
+                    this.removeLabelsFromNode(obj, [obj.children.filter(c => c.isLabel && c.layerId === this.id)]);
```

Comment:
the **`traverse`** method runs through the entire tree structure (all tiles and children). So you could remove label in **`traverse`**;

Reply:
Yes indeed: but better, I ditched the traverse because I can limit the children I visit. See my last update !